numpy/lib/tests/test_io.py

```diff
@@ -466,8 +486,135 @@ def write(self, text):
+                with temppath(suffix=suffix) as name:
+                    with fopen(name, mode='wt', encoding='UTF-32-LE') as f:
+                        f.write(data)
+                    res = getattr(np, self.loadfunc)(name,
+                                                     encoding='UTF-32-LE')
+                    assert_array_equal(res, wanted)
+                    res = getattr(np, self.loadfunc)(
+                                 fopen(name, "rt", encoding='UTF-32-LE'))
+                    assert_array_equal(res, wanted)
+
+    # Python2 .open does not support encoding
+    @np.testing.dec.skipif(MAJVER == 2)
+    def test_compressed_gzip(self):
+        self.check_compressed(gzip.open, ('.gz',))
+
+    @np.testing.dec.skipif(MAJVER == 2 or not HAS_BZ2)
+    def test_compressed_gzip(self):
+        self.check_compressed(bz2.open, ('.bz2',))
+
+    @np.testing.dec.skipif(MAJVER == 2 or not HAS_LZMA)
+    def test_compressed_gzip(self):
+        self.check_compressed(lzma.open, ('.xz', '.lzma'))
+
+    def test_encoding(self):
+        with temppath() as path:
+            with open(path, "wb") as f:
+                f.write('0.\n1.\n2.'.encode("UTF-16"))
+            x = getattr(np, self.loadfunc)(path, encoding="UTF-16")
+            assert_array_equal(x, [0., 1., 2.])
+
+    def test_stringload(self):
+        # umlaute
+        nonascii = b'\xc3\xb6\xc3\xbc\xc3\xb6'.decode("UTF-8")
+        with temppath() as path:
+            with open(path, "wb") as f:
+                f.write(nonascii.encode("UTF-16"))
+            x = getattr(np, self.loadfunc)(path, encoding="UTF-16", dtype=np.unicode)
+            assert_array_equal(x, nonascii)
+
+    def test_binary_decode(self):
+        utf16 = b'\xff\xfeh\x04 \x00i\x04 \x00j\x04'
+        v = getattr(np, self.loadfunc)(BytesIO(utf16), dtype=np.unicode,
+                                       encoding='UTF-16')
+        assert_array_equal(v, np.array(utf16.decode('UTF-16').split()))
+
+    def test_converters_decode(self):
+        # test converters that decode strings
+        c = TextIO()
+        c.write(b'\xcf\x96')
+        c.seek(0)
+        x = getattr(np, self.loadfunc)(c, dtype=np.unicode,
+                       converters={0: lambda x: x.decode('UTF-8')})
+        a = np.array([b'\xcf\x96'.decode('UTF-8')])
+        assert_array_equal(x, a)
+
+    def test_converters_nodecode(self):
+        # test native string converters enabled by setting an encoding
+        utf8 = b'\xcf\x96'.decode('UTF-8')
+        with temppath() as path:
+            with io.open(path, 'wt', encoding='UTF-8') as f:
+                f.write(utf8)
+            x = getattr(np, self.loadfunc)(path, dtype=np.unicode,
+                                           converters={0: lambda x: x + 't'},
+                                           encoding='UTF-8')
+            a = np.array([utf8 + 't'])
+            assert_array_equal(x, a)
+
+
+class TestLoadTxt(LoadTxtBase):
+    loadfunc = 'loadtxt'
```

Comment:
I think this would be clearer as simply `loadfunc = np.loadtxt`

Reply:
It was a workaround for missing `staticmethod`, so this needs to be `loadfunc = staticmethod(np.loadtxt)`